alexawp.php

```diff
@@ -0,0 +1,299 @@
+<?php
+/**
+ * Plugin Name: Alexa WP
+ * Description: Create Alexa skills using your WordPress site
+ * Plugin URI: https://github.com/tomharrigan/
+ * Author: TomHarrigan
+ * Author URI: https://alexawp.com
+ * Version: 0.2
+ * Text Domain: alexawp
+ * License: MIT
+ */
+
+define( 'ALEXAWP_PATH', dirname( __FILE__ ) );
+
+register_activation_hook( __FILE__, 'alexawp_activate' );
+function alexawp_activate() {
+	Alexawp::get_instance();
+	flush_rewrite_rules();
+}
+
+register_deactivation_hook( __FILE__, 'alexawp_deactivate' );
+function alexawp_deactivate() {
+	flush_rewrite_rules();
+}
+
+/**
+ * Compatibility requirements.
+ */
+require_once( ALEXAWP_PATH . '/compat.php' );
+
+/**
+ * Post Type Base Class
+ */
+require_once( ALEXAWP_PATH . '/post-types/class-alexawp-post-type.php' );
+
+/**
+ * Skill Post Type
+ */
+require_once( ALEXAWP_PATH . '/post-types/class-alexawp-post-type-skill.php' );
+
+/**
+ * Flash Briefing Post Type
+ */
+require_once( ALEXAWP_PATH . '/post-types/class-alexawp-post-type-briefing.php' );
+
+/**
+ * Fieldmanager custom fields
+ */
+function alexawp_load_fieldmanager_fields() {
+	require_once( ALEXAWP_PATH . '/fields.php' );
+}
+add_action( 'init', 'alexawp_load_fieldmanager_fields' );
+
+/**
+ * Load a class from within the plugin based on a class name.
+ *
+ * @param string $classname Class name to load.
+ */
+function alexawp_autoload_function( $classname ) {
+	if ( class_exists( $classname ) || 0 !== strpos( $classname, 'Alexa' ) ) {
+		return;
+	}
+	$class = str_replace( '\\', DIRECTORY_SEPARATOR, str_replace( '_', '-', strtolower( $classname ) ) );
+
+	// create the actual filepath
+	$file_path = ALEXAWP_PATH . DIRECTORY_SEPARATOR . $class . '.php';
+
+	// check if the file exists
+	if ( file_exists( $file_path ) ) {
+		// require once on the file
+		require_once $file_path;
+	}
+}
+
+/**
+ * Get the post types whose content is included in the bundled News skill.
+ *
+ * @return array Post type names.
+ */
+function alexawp_news_post_types() {
+	/**
+	 * Filters the post types whose content is included in the bundled News skill.
+	 *
+	 * @param array $post_types Post type names.
+	 */
+	return apply_filters( 'alexawp_post_types', array( 'post' ) );
+}
+
+/**
+ * Get the taxonomies whose terms can be specified by users invoking the News skill.
+ *
+ * @return array Taxonomy names.
+ */
+function alexawp_news_taxonomies() {
+	$option = get_option( 'alexawp-settings' );
+
+	$taxonomies = ( empty( $option['latest_taxonomies'] ) ) ? array() : $option['latest_taxonomies'];
+
+	// Nonexistant taxonomies can shortcircuit get_terms().
+	return array_filter( $taxonomies, 'taxonomy_exists' );
+}
+
+spl_autoload_register( 'alexawp_autoload_function' );
+
+add_action( 'init', array( 'Alexawp', 'get_instance' ), 0 );
+
+use Alexa\Request\IntentRequest;
```

Comment:
I might move here below into it's own class file. This is getting a little cluttered for a main plugin file.